### PR TITLE
Upgrade to node 16

### PR DESCRIPTION
**Pull request by axrj**
Resolves https://github.com/mapbox/vtshaver/issues/54

- Upgrade node to v16
- Removes cxx11abi flag which is now the default for compilers building with latest libc++
- Uses mbgl-core library @ 1.6.0-cxx11abi built with new compiler to remove flag conflict while building vtshaver
- Updates dependencies and devDependencies to newer versions

#### Benchmarking with this branch
Current master benchmarking was done [here](https://github.com/mapbox/vtshaver/pull/55#issuecomment-1042970908) for the merged branch (axrj/publish-napi) 
```
22-02-25 15:23 - axrj mapbox/vtshaver [axrj/node-16] % node bench/bench-batch.js --iterations 50 --concurrency 10
Running benchmark...
Benchmark speed: 45455 runs/s (runs:1500 ms:33 )
Note: pass --mem to track memory usage
Benchmark iterations: 50 concurrency: 10
Success: rate(45454.545454545456 ops/s) > min_rate(1000)
22-02-25 15:23 - axrj mapbox/vtshaver [axrj/node-16] % node bench/bench-batch.js --iterations 50 --concurrency 10
Running benchmark...
Benchmark speed: 50000 runs/s (runs:1500 ms:30 )
Note: pass --mem to track memory usage
Benchmark iterations: 50 concurrency: 10
Success: rate(50000 ops/s) > min_rate(1000)
22-02-25 15:23 - axrj mapbox/vtshaver [axrj/node-16] % node bench/bench-batch.js --iterations 50 --concurrency 10
Running benchmark...
Benchmark speed: 51724 runs/s (runs:1500 ms:29 )
Note: pass --mem to track memory usage
Benchmark iterations: 50 concurrency: 10
Success: rate(51724.13793103448 ops/s) > min_rate(1000)
22-02-25 15:23 - axrj mapbox/vtshaver [axrj/node-16] % node bench/bench-batch.js --iterations 50 --concurrency 10
Running benchmark...
Benchmark speed: 50000 runs/s (runs:1500 ms:30 )
Note: pass --mem to track memory usage
Benchmark iterations: 50 concurrency: 10
Success: rate(50000 ops/s) > min_rate(1000)
```

### Next Steps
- [x] Publish binary to s3 with production version
- [ ] Publish npm package
- [ ] Merge to master

### CODE REVIEW

**Comment by axrj** [comment]
zenhub link : https://app.zenhub.com/workspaces/static-api-team-5fcfe278477430002143372a/issues/mapbox/vtshaver/54

File: README.md [hunk 1]
```diff
@@ -68,16 +68,28 @@ Example:
...
 
 ```
-git submodule update --init
+make
 ```
 
-Build binaries
+For Mac M1 users, there are a couple of extra steps before building
 
+- Patch binding.gyp in the local repo clone to comment out linking instructions as follows
```
Comment (v-hal) [inline README.md:79]:
```suggestion
- Comment out linking instructions in your local binding.gyp as follows
```

File: README.md [hunk 2]
```diff
@@ -68,16 +68,28 @@ Example:
...
 ```
-make
+#  'make_global_settings': [
+#    ['CXX', '<(module_root_dir)/mason_packages/.link/bin/clang++'],
+#    ['CC', '<(module_root_dir)/mason_packages/.link/bin/clang'],
+#    ['LINK', '<(module_root_dir)/mason_packages/.link/bin/clang++'],
+#    ['AR', '<(module_root_dir)/mason_packages/.link/bin/llvm-ar'],
+#    ['NM', '<(module_root_dir)/mason_packages/.link/bin/llvm-nm']
+#  ],
+```
```
Comment (v-hal) [inline README.md:88]: Not sure if this is too verbose here, but GH supports diffs in markdown, which is pretty neat:

```diff
-  'make_global_settings': [
-    ['CXX', '<(module_root_dir)/mason_packages/.link/bin/clang++'],
-    ['CC', '<(module_root_dir)/mason_packages/.link/bin/clang'],
-    ['LINK', '<(module_root_dir)/mason_packages/.link/bin/clang++'],
-    ['AR', '<(module_root_dir)/mason_packages/.link/bin/llvm-ar'],
-    ['NM', '<(module_root_dir)/mason_packages/.link/bin/llvm-nm']
-  ],
+  # 'make_global_settings': [
+  #   ['CXX', '<(module_root_dir)/mason_packages/.link/bin/clang++'],
+  #   ['CC', '<(module_root_dir)/mason_packages/.link/bin/clang'],
+  #   ['LINK', '<(module_root_dir)/mason_packages/.link/bin/clang++'],
+  #   ['AR', '<(module_root_dir)/mason_packages/.link/bin/llvm-ar'],
+  #   ['NM', '<(module_root_dir)/mason_packages/.link/bin/llvm-nm']
+  # ],
```

Comment (axrj) [reply README.md:88]: yeah, it's nice but it makes it a little verbose for a generic markdown editor like vi or vscode where I usually read Readme's, so I kept it concise.

File: .travis.yml [hunk 3]
```diff
@@ -1,12 +1,13 @@
...
 
-node_js: 12
+node_js: 16
 
 # enable c++11/14 builds
 addons:
   apt:
     sources: [ 'ubuntu-toolchain-r-test' ]
-    packages: [ 'libstdc++-5-dev' ]
+    packages: [ 'libstdc++-5-dev', 'python3.6' ]
```
Comment (OzQu) [inline .travis.yml:10]: Probably not needed to have 'python3.6' separately, since you use bionic. Check this.

File: mason-versions.ini [hunk 4]
```diff
@@ -8,4 +8,4 @@ clang-tidy=10.0.0
 clang-format=10.0.0
 llvm-cov=10.0.0
 binutils=2.31
-mbgl-core=1.6.0
\ No newline at end of file
+mbgl-core=1.6.0-cxx11abi
```
Comment (OzQu) [inline mason-versions.ini:11]: Is 1.6.0-cxx11abi the final version, or some development version?

Comment (axrj) [reply mason-versions.ini:11]: It is indeed the final production version. It is tagged that way to be able to identify as there are no other changes apart from being rebuilt with new compiler.

File: .travis.yml [hunk 5]
```diff
@@ -1,12 +1,13 @@
 language: node_js
+dist: bionic
 
-node_js: 12
+node_js: 16
```
[the remark this answers is not in the document]
Comment (axrj) [reply .travis.yml]: Done

File: .travis.yml (hunk 3 above)
Comment (axrj) [reply .travis.yml]: Done.

**Review by OzQu — APPROVED** [review]
Looking good to me!

File: common.gypi [hunk 6]
```diff
@@ -1,17 +1,7 @@
...
-      '-std=c++14',
-      # The assumption is that projects based on node-cpp-skel will also
-      # depend on mason packages. Currently (this will change in future mason versions)
-      # mason packages default to being built/linked with the CXX11_ABI=0.
-      # So we need to link this way too. This allows source
-      # compiling your module on any ubuntu version, even the latest
-      # where the CXX11_ABI default has flipped to 1
-      # More details at https://github.com/mapbox/mason/issues/319
-      '-D_GLIBCXX_USE_CXX11_ABI=0'
-    ],
```
Comment (sofita) [inline common.gypi:14]: @johnk1 Just looking through this and seeing this flag was removed, and wondering if it seems like it may be correlated?

**Pull request merged**